modules/distribution-workflowoperation/src/main/java/org/opencastproject/workflow/handler/distribution/ConfigurablePublishWorkflowOperationHandler.java

```diff
@@ -124,6 +136,36 @@ protected DownloadDistributionService getDistributionService() {
     return distributionService;
   }
 
+  private void addMetadataCatalogsToTemplateData(Map<String, Object> values, MediaPackage mp) {
+    if (! mp.hasCatalogs()) {
+      logger.info("No metdata catalogs found.");
+      return;
+    }
+    if (values == null) values = new HashMap<>();
+
+    for (Catalog catalog : mp.getCatalogs()) {
+      if (catalog.getFlavor() != null) {
+        String mainflavor = catalog.getFlavor().getType();
+        String subflavor = catalog.getFlavor().getSubtype();
+        Map<String, Object> metadata = new HashMap<>();
+        try {
+          DublinCoreCatalog dc = DublinCoreUtil.loadDublinCore(workspace, catalog);
+          for (Map.Entry<EName, List<DublinCoreValue>> entry : dc.getValues().entrySet()) {
+            String value = entry.getValue().get(0).getValue();
+            metadata.put(entry.getKey().getLocalName(), value);
+          }
+          if (!values.containsKey(mainflavor)) {
+            values.put(mainflavor, new HashMap<>());
+          }
+          ((Map<String, Object>)values.get(mainflavor)).put(subflavor, metadata);
+        } catch (Exception e) {
+          logger.error("Catalog from URL %s with flavor %s could not be loaded", catalog.getURI(),
```

Comment:
```suggestion
          logger.error("Catalog from URL {} with flavor {}/{} could not be loaded", catalog.getURI(),
                  mainflavor, subflavor);
```

But wouldn't it make more sense to not catch the exception and let the operation fail if such an error occurs?

Reply:
I don't think that failing is an option. We are looking for metadata catalogs to look for the key-value pairs there. If a catalog is not available for whatever reason we might have corrupt metadata, but in most cases that must not have any consequences why the publishing should not still work.